src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,54 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+	
+    public String value;
+    protected boolean isPrivate;
+
+    public Contact(String contact, boolean isPrivate, 
+    		String MESSAGE_CONTACT_CONSTRAINTS , String iden)throws IllegalValueException{
+    	this.isPrivate = isPrivate;
+        String trimmedContact = contact.trim();
+        if(iden.equals("idenPhone")){
```

Comment:
Note that by doing this, you have caused the parent class (Contact) to have logic to deal with its child classes. This should not be the case as we want to allow child classes to extend this parent class without the need to modify the parent class all the time.